sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala

```diff
@@ -206,22 +206,22 @@ private[sql] case class PreWriteCheck(conf: SQLConf, catalog: SessionCatalog)
         // The relation in l is not an InsertableRelation.
         failAnalysis(s"$l does not allow insertion.")
 
-      case c: CreateTableUsingAsSelect =>
+      case CreateTable(tableDesc, mode, Some(query)) =>
```

Comment:
The exception is CREATE HIVE TABLE AS SELECT. So far, we assume all the unsupported cases have been captured in the DDL handling. Thus, I think it is safe to apply all the cases to Hive Tables.